services/src/main/java/org/keycloak/protocol/ciba/resolvers/DefaultCIBALoginUserResolver.java

```diff
@@ -1,19 +1,35 @@
 package org.keycloak.protocol.ciba.resolvers;
 
+import org.apache.commons.lang.StringUtils;
+import org.jboss.logging.Logger;
+import org.keycloak.crypto.Aes128GcmEncryptor;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.UserModel;
 import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.protocol.ciba.decoupledauthn.DelegateDecoupledAuthenticationProvider;
+
+import java.security.GeneralSecurityException;
 
 public class DefaultCIBALoginUserResolver implements CIBALoginUserResolver {
 
     private KeycloakSession session;
+    private static final Logger logger = Logger.getLogger(DelegateDecoupledAuthenticationProvider.class);
 
     public DefaultCIBALoginUserResolver(KeycloakSession session) {
         this.session = session;
     }
 
     @Override
     public UserModel getUserFromLoginHint(String loginHint) {
+        String secret = session.getContext().getClient().getSecret();
+        if (StringUtils.isNotBlank(secret)) {
```

Comment:
It causes the failures of all tests of existing arquillian integration test `CIBATest`.

IMO, it might to be better to add the Client setting `login_hint encryption` switch.

### Where to add new setting item on Admin Console
IMO, on current client's "Settings" UI, add this `login_hint encryption` switch on existing "Advanced Settings" pain.

### Where to look if we  introduce new client setting 

Sometimes we need to introduce new Client setting.

There are two types of Client setting.

- OAuth/OIDC standard document specified Client Metadata
  - OAuth2
     https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml
  - OIDC
     https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata
- NOT OAuth/OIDC standard document specified Client Metadata

AFAIK, entities we need to look at when we introduce new client setting (regardless of whether it is OAuth/OIDC standard document specified Client Metadata or not) are as follows.

- Representation
 It stands for the format in representing Client.
 This format is used when communicating between keycloak and other entities.
   - [1] ClientRepresentation
 It is Client's Representation.
 When we access Client by Admin REST API(namely, on Admin Console), this ClientRepresentation is used.

- Model
 It stands for the format used when keycloak's internal program treats Client.
  - [2]  ClientModel
 It is Client's Model

- Store
 It stands for the format used when keycloak's internal program stores Client.
   - Infinispan Cache
  Used when storing Client in memory
      - [3] ClientAdapter(infinispan)
   - JPA:
  Used when storing Client in persistent storage
      - [4] ClientAdapter(jpa)
      - [5] ClientEntity(jpa)

- OIDC Extension
 It extends Client so that it can be treated as OIDC conformant Client
   - Representation
     - [6] OIDCClientRepresentation
   It stands for the format in representating OIDC conformant Client.
   This format is used when communicating between keycloak and other entities in OIDC manner
   eg.
   When we access Client by OIDC Dynamic Client Registration, this OIDCClientRepresentation is used.

   - Representation/Model
      - [7] OIDCAdvancedConfigWrapper
    It works as the wrapper between ClientRepresentation and OIDCClientRepresentation
      - [8] DescriptionConverter
    It converts the Client representation between ClientRepresentation and OIDCClientRepresentation.
      - [9] OIDCConfigAttributes
     It define keys used to store/access OIDC related parameter to ClientRepresentation's map object called Attribute and ClientModel's map object also callled Attribute.

- UI
   - [10] client.js
   - [11] client-detail.html
   - [12] admin-messages_en.properties 

Basically,  we need not modify [3][4][5].
We need to modify [6][7] if we introduce new OAuth/OIDC standard document specified Client Metadata.

#### For OAuth/OIDC standard document specified Client Metadata
We need to modify [1][2][6][7][8][9][10][11][12]. 

#### For NOT OAuth/OIDC standard document specified Client Metadata
We need to modify [1][2][8][9][10][11][12].

### Reference
The PRs about newly introducing Client settings are as follows :

- https://github.com/keycloak/keycloak/pull/6581
- https://github.com/keycloak/keycloak/pull/7065

Reply:
So maybe here it would be better just check it for null and not for being blank assuming all our clients should be private and have to have a client_secret?

Reply:
By adding a setting I suppose you mean add new flag to client configuration to check it afterwards from within our code for like if/else statement?

Reply:
Yes, e.g. "CIBA Login Hint Encryption" switch whose boolean value is stored on Client's attribute.

Reply:
Added switch on FE and corresponding checks in BE ([6-12]). Still didn't get why do we need to change [1] & [2], so advice would be highly appreciated.

Reply:
StringUtils removed from code.